ratis-server/src/main/java/org/apache/ratis/server/impl/StateMachineUpdater.java

```diff
@@ -341,6 +364,10 @@ private void notifyAppliedIndex(long index) {
     appliedIndexConsumer.accept(index);
   }
 
+  void setRemoving() {
+    this.isRemoving = true;
+  }
+
```

Comment:
Let's move it to right below `shouldTakeSnapshotAtStop()`.

Reply:
OK